pkgs/development/tools/global-platform-pro/default.nix

```diff
@@ -0,0 +1,89 @@
+{ stdenv, fetchFromGitHub, jdk, maven, writeText, makeWrapper, jre_headless, pcsclite }:
+
+# TODO: This is quite a bit of duplicated logic with gephy. Factor it out?
+stdenv.mkDerivation rec {
+    pname = "global-platform-pro";
+    version = "0.3.10-rc11";
+    describeVersion = "v0.3.10-rc11-0-g8923747"; # git describe --tags --always --long --dirty
+    name = "${pname}-${version}";
+
+    src = fetchFromGitHub {
+        owner = "martinpaljak";
+        repo = "GlobalPlatformPro";
+        rev = "v${version}";
+        sha256 = "0rk81x2y7vx1caxm6wa59fjrfxmjn7s8yxaxm764p8m2qxk3m4y2";
+    };
+
+    patches = [ (writeText "${name}-version.patch" ''
+        diff --git a/pom.xml b/pom.xml
+        index 1e5a82d..1aa01fe 100644
+        --- a/pom.xml
+        +++ b/pom.xml
+        @@ -121,14 +121,10 @@
+                             </execution>
+                         </executions>
+                         <configuration>
+        -                    <executable>git</executable>
+        +                    <executable>echo</executable>
+                             <outputFile>target/generated-resources/pro/javacard/gp/pro_version.txt</outputFile>
+                             <arguments>
+        -                        <argument>describe</argument>
+        -                        <argument>--tags</argument>
+        -                        <argument>--always</argument>
+        -                        <argument>--long</argument>
+        -                        <argument>--dirty</argument>
+        +                        <argument>${describeVersion}</argument>
+                             </arguments>
+                         </configuration>
+                     </plugin>
+    '') ];
+
+    deps = stdenv.mkDerivation {
+        name = "${name}-deps";
+        inherit src patches;
+        nativeBuildInputs = [ jdk maven ];
+        buildPhase = ''
+            while mvn package -Dmaven.repo.local=$out/.m2 -Dmaven.wagon.rto=5000; [ $? = 1 ]; do
+                echo "timeout, restart maven to continue downloading"
+            done
+        '';
+        # keep only *.{pom,jar,sha1,nbm} and delete all ephemeral files with lastModified timestamps inside
+        installPhase = ''find $out/.m2 -type f -regex '.+\(\.lastUpdated\|resolver-status\.properties\|_remote\.repositories\)' -delete'';
```

Comment:
(damn that is an ugly regex syntax)

Please quote the shell variable and maybe put the command into its own line:

```
installPhase = ''
# command
''
```

Where is this actually installing anything?

Reply:
Done, and split the line :) (even though it's `$out` so it's guaranteed to not contain spaces, I do understand what you mean by wanting to always quote variables :))

As for where this is installing anything, my understanding of this code (from `gephy`) is that the `buildPhase` is doing the downloading, and the `installPhase` is doing the cleanup. This is similar to the usual way of doing the build in the `buildPhase` and then copying only what's required in the `installPhase`, so I think it makes more or less sense? :)

Reply:
There probably is room for generalization, but I don't know maven either.